server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/BucketNameResolver.java

```diff
@@ -84,33 +85,39 @@ private BucketNameResolver(Optional<BucketName> namespace, Optional<String> pref
         this.prefix = prefix;
     }
 
-    BucketName resolve(BucketName bucketName) {
+    ResolvedBucketName resolve(BucketName bucketName) {
         Preconditions.checkNotNull(bucketName);
 
         if (isNameSpace(bucketName)) {
-            return bucketName;
+            return ResolvedBucketName.of(bucketName.asString());
         }
         return prefix
-            .map(bucketPrefix -> BucketName.of(bucketPrefix + bucketName.asString()))
-            .orElse(bucketName);
+            .map(bucketPrefix -> ResolvedBucketName.of(bucketPrefix + bucketName.asString()))
+            .orElse(ResolvedBucketName.of(bucketName.asString()));
     }
 
-    Optional<BucketName> unresolve(BucketName bucketName) {
+    Optional<BucketName> unresolve(ResolvedBucketName bucketName) {
         if (isNameSpace(bucketName)) {
-            return Optional.of(bucketName);
+            return Optional.of(BucketName.of(bucketName.asString()));
         }
 
         return prefix.map(p -> {
             if (bucketName.asString().startsWith(p)) {
                 return Optional.of(BucketName.of(bucketName.asString().substring(p.length())));
             }
             return Optional.<BucketName>empty();
-        }).orElse(Optional.of(bucketName));
+        }).orElse(Optional.of(BucketName.of(bucketName.asString())));
     }
 
     private boolean isNameSpace(BucketName bucketName) {
         return namespace
-            .map(existingNamespace -> existingNamespace.equals(bucketName))
+            .map(existingNamespace -> existingNamespace.asString().equals(bucketName.asString()))
```

Comment:
**💭 thought:** the namespace term is also called defaultBucketName in other places ( it is called namespace in the configuration file, the configuration builder accepts a namespace setter but when building the actual configuration object it injects the namespace into a field named `defaultBucketName` a similarly named field is also exposed by the S3 dao but in the bucketname resolver it is again  called namespace. It is unclear to me what this concept is supposed to represent and how it is different from bucketPrefix (whose name is fairly stable throughout the code)
I don't expect an action on this MR but seeing the name here reminded me of this thought.

Reply:
**❔ question:** ‏why change the equality from `BucketName#equals` to `asString().equals` I agree that the result is identical but don't think this change makes the code clearer

I understand why it is necessary in the  overload of `isNameSpace` that compares a `ResolvedBucketName` and a `BucketName`.